src/iosrv/init.c

```diff
@@ -578,6 +578,11 @@ server_init(int argc, char *argv[])
 		goto exit_drpc_fini;
 	D_INFO("Modules successfully set up\n");
 
+	rc = dss_srv_mark_xstreams_up();
```

Comment:
I didn't know that. After trying to use dss_xstreams_open_barrier, I realize it's not as simple as calling it later after module setup. Because we still need to use it to wait for all xstreams are ready, before calling module setup callbacks, as they may need to do collective operations.

Another potential concern is this conditional variable is not selective. It blocks all RPCs, including CaRT internal ones that do not call back to DAOS. (And all dRPCs, if we need to block any. But certain dRPCs are needed for bootstrapping.)

Reply:
After some discussions, it seems blocking all RPCs is preferred by most.